From f27dc29a78dca3356f7a55f31a37079b60509847 Mon Sep 17 00:00:00 2001
From: "guoping.li" <guoping.li@amlogic.com>
Date: Tue, 16 May 2023 16:32:10 +0800
Subject: [PATCH 2/8] tvin: tvin default port [5/5]

PD#SWPL-122352

Problem:
gstreamer plugin add tvin property, it's not easy to use

Solution:
add default tvin value:
HDMI RX, default is 0, video layer
Screen recorder, default is 0x110000001, vpp0 osd+video layer

Verify:
local

Signed-off-by: guoping.li <guoping.li@amlogic.com>
Change-Id: I563deea0b01d81206a8c8df8de9a7af45e5058e1
---
 ...l2src-patch-for-support-mediactrlsrc.patch | 25 +++++++++----------
 ...l2src-patch-for-support-mediactrlsrc.patch | 22 ++++++++--------
 2 files changed, 23 insertions(+), 24 deletions(-)
 mode change 100755 => 100644 recipes-multimedia/gstreamer/gst1-plugins-good/00029-v4l2src-patch-for-support-mediactrlsrc.patch

diff --git a/recipes-multimedia/gstreamer/gst1-plugins-good-1.20.3/0004-v4l2src-patch-for-support-mediactrlsrc.patch b/recipes-multimedia/gstreamer/gst1-plugins-good-1.20.3/0004-v4l2src-patch-for-support-mediactrlsrc.patch
index de390103..ce1436cd 100755
--- a/recipes-multimedia/gstreamer/gst1-plugins-good-1.20.3/0004-v4l2src-patch-for-support-mediactrlsrc.patch
+++ b/recipes-multimedia/gstreamer/gst1-plugins-good-1.20.3/0004-v4l2src-patch-for-support-mediactrlsrc.patch
@@ -1,9 +1,8 @@
-From 827e0ad10bdc0decac1b5819f85cf9894b6a1087 Mon Sep 17 00:00:00 2001
+From b7624dcfafdaaf5dfcc1cb5112f3019ba1553b61 Mon Sep 17 00:00:00 2001
 From: "guoping.li" <guoping.li@amlogic.com>
-Date: Fri, 12 May 2023 17:51:27 +0800
-Subject: [PATCH] v4l2src: patch for support mediactrlsrc  [1/1]
+Date: Tue, 16 May 2023 15:06:27 +0800
+Subject: [PATCH] v4l2src-patch-for-support-mediactrlsrc
 
-Signed-off-by: guoping.li <guoping.li@amlogic.com>
 ---
  .../gst-plugins-good/sys/v4l2/gstv4l2src.c     | 18 +++++++++++++++++-
  .../gst-plugins-good/sys/v4l2/meson.build      |  4 +++-
@@ -11,7 +10,7 @@ Signed-off-by: guoping.li <guoping.li@amlogic.com>
  3 files changed, 27 insertions(+), 2 deletions(-)
 
 diff --git a/subprojects/gst-plugins-good/sys/v4l2/gstv4l2src.c b/subprojects/gst-plugins-good/sys/v4l2/gstv4l2src.c
-index b26ddba..f15f159 100755
+index b26ddba8..f15f1591 100755
 --- a/subprojects/gst-plugins-good/sys/v4l2/gstv4l2src.c
 +++ b/subprojects/gst-plugins-good/sys/v4l2/gstv4l2src.c
 @@ -896,6 +896,10 @@ gst_v4l2src_stop (GstBaseSrc * src)
@@ -51,7 +50,7 @@ index b26ddba..f15f159 100755
      default:
        break;
 diff --git a/subprojects/gst-plugins-good/sys/v4l2/meson.build b/subprojects/gst-plugins-good/sys/v4l2/meson.build
-index 65f551f..9e7322e 100755
+index 65f551fb..9e7322ed 100755
 --- a/subprojects/gst-plugins-good/sys/v4l2/meson.build
 +++ b/subprojects/gst-plugins-good/sys/v4l2/meson.build
 @@ -59,12 +59,14 @@ if have_v4l2
@@ -71,12 +70,12 @@ index 65f551f..9e7322e 100755
      install_dir : plugins_install_dir,
    )
 diff --git a/subprojects/gst-plugins-good/sys/v4l2/v4l2_calls.c b/subprojects/gst-plugins-good/sys/v4l2/v4l2_calls.c
-index 94c1682..566aeed 100755
+index 94c1682c..ce6d20b4 100755
 --- a/subprojects/gst-plugins-good/sys/v4l2/v4l2_calls.c
 +++ b/subprojects/gst-plugins-good/sys/v4l2/v4l2_calls.c
-@@ -581,6 +581,13 @@ gst_v4l2_open (GstV4l2Object * v4l2object, GstV4l2Error * error)
- 
-   gst_v4l2_adjust_buf_type (v4l2object);
+@@ -538,6 +538,13 @@ gst_v4l2_open (GstV4l2Object * v4l2object, GstV4l2Error * error)
+   if (!S_ISCHR (st.st_mode))
+     goto no_device;
  
 +  // get default tvin port for device
 +  // screen : OSD+Video
@@ -85,9 +84,9 @@ index 94c1682..566aeed 100755
 +    v4l2object->tvin_port=get_default_tvin_port(&v4l2object->videodev);
 +  }
 +
-   if (v4l2object->tvin_port != -1) {
-     unsigned int portType = v4l2object->tvin_port;
-     if (v4l2object->ioctl(v4l2object->video_fd, VIDIOC_S_INPUT, &portType) < 0) {
+   /* open the device */
+   v4l2object->video_fd =
+       open (v4l2object->videodev, O_RDWR /* | O_NONBLOCK */ );
 -- 
 2.25.1
 
diff --git a/recipes-multimedia/gstreamer/gst1-plugins-good/00029-v4l2src-patch-for-support-mediactrlsrc.patch b/recipes-multimedia/gstreamer/gst1-plugins-good/00029-v4l2src-patch-for-support-mediactrlsrc.patch
old mode 100755
new mode 100644
index 77eb4140..f884fc3a
--- a/recipes-multimedia/gstreamer/gst1-plugins-good/00029-v4l2src-patch-for-support-mediactrlsrc.patch
+++ b/recipes-multimedia/gstreamer/gst1-plugins-good/00029-v4l2src-patch-for-support-mediactrlsrc.patch
@@ -1,6 +1,6 @@
-From c160db221dbb21a3951b683b487390f291ae7516 Mon Sep 17 00:00:00 2001
+From d0951eb07f67cb0bbe78cde0773b685a4e91f72d Mon Sep 17 00:00:00 2001
 From: "guoping.li" <guoping.li@amlogic.com>
-Date: Fri, 12 May 2023 23:54:59 +0800
+Date: Tue, 16 May 2023 14:33:20 +0800
 Subject: [PATCH] v4l2src-patch-for-support-mediactrlsrc
 
 ---
@@ -87,23 +87,23 @@ index ce591b3..5704c51 100644
      install_dir : plugins_install_dir,
    )
 diff --git a/sys/v4l2/v4l2_calls.c b/sys/v4l2/v4l2_calls.c
-index 13a0189..d3a18b3 100644
+index 13a0189..379d80d 100644
 --- a/sys/v4l2/v4l2_calls.c
 +++ b/sys/v4l2/v4l2_calls.c
-@@ -617,6 +617,13 @@ gst_v4l2_open (GstV4l2Object * v4l2object)
- 
-   gst_v4l2_adjust_buf_type (v4l2object);
+@@ -574,6 +574,13 @@ gst_v4l2_open (GstV4l2Object * v4l2object)
+   if (!S_ISCHR (st.st_mode))
+     goto no_device;
  
 +  // get default tvin port for device
 +  // screen : OSD+Video
 +  // HDMI RX : Video
 +  if (v4l2object->tvin_port == -1) {
-+    v4l2object->tvin_port=get_default_tvin_port(&v4l2object->videodev);
++    v4l2object->tvin_port = get_default_tvin_port(&v4l2object->videodev);
 +  }
-+  
-   if (v4l2object->tvin_port != -1) {
-     unsigned int portType = v4l2object->tvin_port;
-     if (v4l2object->ioctl(v4l2object->video_fd, VIDIOC_S_INPUT, &portType) < 0) {
++
+   /* open the device */
+   v4l2object->video_fd =
+       open (v4l2object->videodev, O_RDWR /* | O_NONBLOCK */ );
 -- 
 2.25.1
 
-- 
2.25.1

